sonar-plugin-api/src/main/java/org/sonar/api/batch/SensorContext.java

```diff
@@ -191,25 +190,11 @@
 
   // ----------- DEPENDENCIES BETWEEN RESOURCES --------------
 
-  Dependency saveDependency(Dependency dependency);
-
-  /**
-   * @deprecated since 5.1 Sensors should not read but only save data
-   */
-  @Deprecated
-  Set<Dependency> getDependencies();
-
   /**
-   * @deprecated since 5.1 Sensors should not read but only save data
+   * @deprecated since 5.2 No more design features. No-op
    */
   @Deprecated
-  Collection<Dependency> getIncomingDependencies(Resource to);
-
-  /**
-   * @deprecated since 5.1 Sensors should not read but only save data
-   */
-  @Deprecated
-  Collection<Dependency> getOutgoingDependencies(Resource from);
+  Dependency saveDependency(Dependency dependency);
```

Comment:
![INFO](https://raw.githubusercontent.com/henryju/image-hosting/master/info.png)Do not forget to remove this deprecated code someday.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)